pkgs/applications/audio/librespot/default.nix

```diff
@@ -44,13 +46,15 @@ rustPlatform.buildRustPackage rec {
   buildInputs =
     [ openssl ]
     ++ lib.optional withALSA alsa-lib
-    ++ lib.optional withAvahi avahi-compat
```

Comment:
When building with `withAvahi`, the build run fines without `avahi-compat` but breaks when missing with `withDNS-SD=true;`. This also reflects the documentation of `librespot`:

https://github.com/librespot-org/librespot/blob/dev/COMPILING.md#zeroconf-library-dependencies